connector/queue/model_view.xml

```diff
@@ -140,6 +140,21 @@
             <field name="search_view_id" ref="view_queue_job_search"/>
         </record>
 
+        <record id="action_button_done_queue_job" model="ir.actions.server">
+            <field name="name">Set to 'Done'</field>
+            <field name="model_id" ref="model_queue_job"/>
+            <field name="state">code</field>
+            <field name="code"><![CDATA[
+self.button_done(cr, uid, context['active_ids'], context=context)
```

Comment:
@zakiuu uyou should modify the `button_done` method https://github.com/acsone/connector/blob/4d21e60e250a5bd9111b2808c24f7610264251cb/connector/queue/model.py#L133 to filter out jobs not into the expected state.
```python
 @api.multi
    def button_done(self):
        result = _('Manually set to done by %s') % self.env.user.name
        self.filtered(lambda j: j.state in (PENDING, ENQUEUED, FAILED)._change_job_state(DONE, result=result)
        return True
```